### ISSUE TITLE
Spawns, pokestops, outside access..?

### ISSUE THREAD

**Issue by d0dUxDJ** [maierfelix/POGOserver#232]
I'm sorry, I might be missing something but for the life of me I can't find the right documentation if this is written somewhere. I set up the server and it does work nicely with my account (I wonder if I have to take more steps to let other people in other than port forwarding?), I set up a couple a gym and that worked fine. But how do you set pokestop rewards properly? I see there's an "xp" setting, which is clearly a number, but then rewards. Is there an itemlist with IDs? Do I have to put in "code" itemnames? How do I separate values in that cell?

My other issue, spawns. How do I fill them to work properly?

Any clues on all of this would be highly appreciated. Or just point me to the correct docs because I really cannot navigate this wiki, I can't find these infos if they are present. Again, sorry if that's the case.

Thanks in advance!

**Comment by Badley187** [comment]
http://pastebin.com/rs9AWYKq

i have make this for you ;) check this

**Comment by Cronick** [comment]
@Badley187 - There is a few mistakes in the document, but it's not incorrect at some points.

**Comment by Badley187** [comment]
its working for me :) maybe you can correct it and add it to the wiki?
i´m not a programmer but i learning by my self :)

**Comment by Cronick** [comment]
2 = Greatball and 3 = Ultraball, else the document is fine.
And it was not a bad idea to include a wiki page for that.

**Comment by Badley187** [comment]
sorry i´m german :D i forgot to change the names^^ i can try to make a wiki but my english are poor...

**Comment by d0dUxDJ** [comment]
Thanks a bunch @Badley187! So I'm not crazy yet, all of this is not present in the docs of the wiki right?
I can help correcting the eventual grammar mistakes and add those to a wiki, possibly.

**Comment by Badley187** [comment]
you have a go for the wiki :)
i think its in seconds

**Comment by d0dUxDJ** [comment]
I inserted a spawn near me. Now nothing works. API doesn't show anything anymore, and this shows in the terminal:

`(node:8601) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 8): SyntaxError: Unexpected number in JSON at position 1`

![screenshot from 2016-09-28 15-05-51](https://cloud.githubusercontent.com/assets/11587540/18914370/2ce60a4c-858d-11e6-952e-e2eb1c880214.png)

what's wrong?

**Comment by nliakm** [comment]
hm maybe im wrong, but i think, you are not allowed to put more than one pokemon into that spawnpoint. maybe try to replace {150, 151} with 150?

**Comment by d0dUxDJ** [comment]
That worked. But the pokemon in game won't trigger a battle (I tap on it, nothing happens). 

Terminal spits out this: 
`[Console] ##### 15:49:11
[Console] ENCOUNTER 27 => 107
[Console] GET_HATCHED_EGGS 9 => 2
[Console] GET_INVENTORY 7 => 85
[Console] CHECK_AWARDED_BADGES 9 => 2
[Console] DOWNLOAD_SETTINGS 42 => 187
[Console] GET_HATCHED_EGGS 9 => 2
[Console] GET_INVENTORY 7 => 85
[Console] CHECK_AWARDED_BADGES 9 => 2
[Console] DOWNLOAD_SETTINGS 42 => 187
`

**Comment by nliakm** [comment]
so the server console does not give you any error messages?

**Comment by d0dUxDJ** [comment]
Exactly, looks like no error is there. But when tapping on the Pokemon in game it won't initiate the battle sequence. Also I noticed that the "level up" to 40 thing pops up every time I log in. Do you know how to stop that?

**Comment by Badley187** [comment]
u need to use 0.35
 and sorry, for spawnpoint need [150,151] not {150,151}
gyms dont work at this time... its all still in development

**Comment by d0dUxDJ** [comment]
Alright, thanks for the answer @Badley187, I'm going to try 0.35 and edit this message later. Yeah using square brackets works, found that out by myself after checking a spawn made with the api. It's a shame regarding gyms, I was doing all of this to play with friends in a "better structured" PoGO. I guess I'll have to wait a bit :) Thanks for the answer, brb testing 0.35 and Pokémon catching.

**Comment by Badley187** [comment]
give @maierfelix some time and he make this server perfect for us :)

**Comment by d0dUxDJ** [comment]
Tried to catch pokemon 151, or Mew. This happened:

[Console] Sent pm0151 to myemail@gmail.com
[Console] ##### 22:35:50
[Console] ENCOUNTER 27 => 107
[Console] CHECK_CHALLENGE 9 => 3
[Console] GET_HATCHED_EGGS 9 => 2
[Console] GET_INVENTORY 7 => 85
[Console] CHECK_AWARDED_BADGES 9 => 2
[Console] DOWNLOAD_SETTINGS 42 => 187
[Console] ##### 22:35:58
[Console] CATCH_POKEMON 20 => 2
[Console] CHECK_CHALLENGE 9 => 3
[Console] GET_HATCHED_EGGS 9 => 2
[Console] GET_INVENTORY 7 => 85
[Console] CHECK_AWARDED_BADGES 9 => 2
[Console] DOWNLOAD_SETTINGS 42 => 187

This when I threw a pokeball

[Console] Error: ER_NO_DEFAULT_FOR_FIELD: Field 'deployed_fort_id' doesn't have a default value

and this after the throw, the ball stopped there (no moving, like when the game lags on Niantic servers)

(node:12475) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): TypeError: Cannot read property 'caughtBy' of null

In this order:
[Console] Sent pm0151 to myemail@gmail.com
[Console] ##### 22:35:50
[Console] ENCOUNTER 27 => 107
[Console] CHECK_CHALLENGE 9 => 3
[Console] GET_HATCHED_EGGS 9 => 2
[Console] GET_INVENTORY 7 => 85
[Console] CHECK_AWARDED_BADGES 9 => 2
[Console] DOWNLOAD_SETTINGS 42 => 187
[Console] ##### 22:35:58
[Console] CATCH_POKEMON 20 => 2
[Console] CHECK_CHALLENGE 9 => 3
[Console] GET_HATCHED_EGGS 9 => 2
[Console] GET_INVENTORY 7 => 85
[Console] CHECK_AWARDED_BADGES 9 => 2
[Console] DOWNLOAD_SETTINGS 42 => 187
[Console] Error: ER_NO_DEFAULT_FOR_FIELD: Field 'deployed_fort_id' doesn't have a default value
(node:12475) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): TypeError: Cannot read property 'caughtBy' of null

Any clues?

I agree, I'm just impatient, I wish I could help development but I'm still a newbie. One day...

**Comment by Badley187** [comment]
check this https://github.com/maierfelix/POGOserver/issues/222

**Comment by d0dUxDJ** [comment]
Thanks! Fiddled around with null values and I came to the conclusion that what made it work was setting all default values to null except the Pokémon details, which are:
id
owner_id
dex_number
cp
stamina
stamina_max
move_1
move_2

Those should not be modified, all the others need to have default value null.

I managed to catch a Mew! So it is indeed working.

The only remaining issues are: every login I keep getting the "level 40" message and the pokestop I made doesn't give me exp beyond the 2000000 (lvl 40) mark. Catching the Mew gave me 400 extra though. So now I have 2000400, but looting pokestops still isn't giving me extra exp.

Any help? Thanks.
